File: tests/fixtures/fn-expr.input.ts
function applyTwice(x: number): number {
    const double = function(n: number): number {
        return n * 2;
    };
    return double(double(x));
}

function makeMessage(prefix: string): string {
    const format = function(text: string): string {
        return prefix + ": " + text;
    };
    return format("hello");
}

File: tests/fixtures/fn-expr.input.rs
fn applyTwice(x: f64) -> f64 {
    let double = |n: f64| -> f64 { n * 2.0 };
    double(double(x))
}

fn makeMessage(prefix: String) -> String {
    let format = |text: String| -> String { prefix + ": " + &text };
    format("hello".to_string())
}
